cumulus_library/apis/loinc.py

```diff
@@ -0,0 +1,99 @@
+"""Class for communicating with the umls API"""
+
+import os
+import pathlib
+
+import requests
+import rich
+
+from cumulus_library import base_utils, errors
+
+BASE_URL = "https://loinc.regenstrief.org/api/v1/"
+
+
+class LoincApi:
+    def __init__(self, *, user: str | None = None, password: str | None = None):
+        """Creates a requests session for future calls
+
+        :keyword user: the username of the loinc user
+        :keyword password: the password of the loinc user
+
+        You can request a Loinc account at https://loinc.org/join/.
+        """
+        if user is None:
+            user = os.environ.get("LOINC_USER")
+            if user is None:
+                raise errors.ApiError("No LOINC user provided")
+        if password is None:
+            password = os.environ.get("LOINC_PASSWORD")
+            if password is None:
+                raise errors.ApiError("No LOINC password provided")
+
+        self.session = requests.Session()
+        self.session.auth = requests.auth.HTTPBasicAuth(user, password)
+
+    def get_all_download_versions(self) -> list:
+        """returns all available versions available for download"""
+        versions = []
+        res = self.session.get(f"{BASE_URL}Loinc/All")
+        if res.status_code == 401:
+            raise errors.ApiError("Invalid LOINC credentials")
+        for record in res.json():
+            versions.append(record["version"])
+        return versions
+
+    def get_download_info(self, version: str | None = None) -> (str, str):
```

Comment:
I think you need(?) to use `tuple[str, str]` as the return type, not sure the parens work (at least my IDE usually flags it)